File: cosmol_viewer/examples/protein_ligand.rs
use cosmol_viewer::parser::mmcif::parse_mmcif;
use cosmol_viewer::parser::sdf::ParserOptions;
use cosmol_viewer::parser::sdf::parse_sdf;
use cosmol_viewer::utils::VisualShape;
use cosmol_viewer::{Scene, Viewer, shapes::Molecules, shapes::Protein};

fn main() {
    let mmcif_data = parse_mmcif(include_str!("../examples/2AMD.cif"), None);
    let prot = Protein::new(mmcif_data).color([0.2, 0.45, 0.6]);

    let ligand_data = parse_sdf(
        include_str!("2amd_ligand.sdf"),
        &ParserOptions {
            keep_h: true,
            multimodel: true,
            onemol: false,
        },
    );
    let ligand = Molecules::new(ligand_data);

    let mut scene = Scene::new();
    scene.recenter(ligand.get_center());
    scene.add_shape(prot, Some("prot"));
    scene.add_shape(ligand, Some("ligand"));

    Viewer::render(&scene, 800.0, 500.0);

    println!("Press Enter to exit...");
    use std::io::{self, Write};
    let _ = io::stdout().flush();
    let _ = io::stdin().read_line(&mut String::new());
}

File: cosmol_viewer/examples/protein_ligand.py
from cosmol_viewer import Scene, Viewer, parse_sdf, parse_mmcif, Molecules, Protein

mmcif_data  = parse_mmcif(open("./examples/2AMD.cif", "r", encoding="utf-8").read())
prot = Protein(mmcif_data).color([0.2, 0.45, 0.6])

ligand_data  = parse_sdf(open("./examples/2AMD_ligand.sdf", "r", encoding="utf-8").read())
ligand = Molecules(ligand_data)

scene = Scene()
scene.add_shape(prot, "prot")
scene.add_shape(ligand, "ligand")
scene.recenter(ligand.get_center())

viewer = Viewer.render(scene, width=800, height=500)

viewer.save_image("screenshot.png")

print("Press Any Key to exit...", end='', flush=True)
_ = input()
